wwwapp/models.py

```diff
@@ -281,3 +305,49 @@ def result_in_percent(self):
 
     class Meta:
         unique_together = [('workshop', 'participant')]
+
+
+class ResourceYearPermission(models.Model):
+    """
+    Resource associated with a WWW edition (year). Resource can be accessed by
+    users who qualified or had a lecture on year. User is granted access to
+    root_url and recursively to all files and subdirectories inside.
+    """
+    display_name = models.CharField(max_length=50, blank=True)
+    access_url = models.URLField(blank=True,
+                                 help_text="URL dla przycisku w menu. Przycisk nie jest wyświetlany jeśli url jest pusty")
+    root_path = models.CharField(max_length=256, null=False, blank=False,
+                                 help_text='bez "/" na końcu. np. "/internety/www15"')
+    year = models.IntegerField(null=False, blank=False)
+
+    def __str__(self):
+        return "{} - {}".format(self.year,
+                                self.display_name if self.display_name != "" else self.root_path)
+
+    def clean(self):
+        super().clean()
+        if self.access_url != "" and self.display_name == "":
+            raise ValidationError("Wyświetlana nazwa musi być ustawiona jeśli "
+                                  "URL dostępu jest ustawiony")
+
+        if self.root_path.endswith("/"):
+            self.root_path = self.root_path[:-1]
+        if not self.root_path.startswith("/"):
+            self.root_path = "/" + self.root_path
+
+    @staticmethod
+    def resources_for_uri(uri: str):
+        scheme, netloc, path, query, fragment = urllib.parse.urlsplit(uri)
+        path = os.path.normpath(path)  # normalize path
+        path_parts = path.split('/')
+
+        query = Q(pk__isnull=True)  # always false
+        for i in range(len(path_parts)):
+            query |= Q(root_path='/'.join(path_parts[:i+1]))
```

Comment:
This code checks the following prefixes:
```
(empty string)
/internet
/internet/www13
```
The first one should be `/`, not empty string.

Reply:
Ok! More IFs INCOMING!!!!